exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentTraceServiceClients.java

```diff
@@ -0,0 +1,280 @@
+      currentConfigObserverRef = new AtomicReference<>();
+
+  // Stream observer for ExportTraceServiceResponse.
+  // This observer is a no-op and won't block the thread.
+  private static final StreamObserver<ExportTraceServiceResponse> exportResponseObserver =
+      new StreamObserver<ExportTraceServiceResponse>() {
+        @Override
+        public void onNext(ExportTraceServiceResponse value) {
+          // Do nothing since ExportTraceServiceResponse is an empty message.
+        }
+
+        @Override
+        public void onError(Throwable t) {}
+
+        @Override
+        public void onCompleted() {}
+      };
+
+  // Tries to establish connections to Agent server with the given endPoint. If succeeded, create
+  // ExportServiceClient and ConfigServiceClientWorker, start ConfigServiceClientWorker in a worker
+  // thread, and return ExportServiceClient. Otherwise return null.
+  @Nullable
+  static ExportServiceClient openStreams(String endPoint, boolean useInsecure, String serviceName) {
+    ManagedChannelBuilder<?> channelBuilder = ManagedChannelBuilder.forTarget(endPoint);
+    if (useInsecure) {
+      channelBuilder.usePlaintext();
+    }
+    ManagedChannel channel = channelBuilder.build();
+    TraceServiceStub stub = TraceServiceGrpc.newStub(channel);
+    Node node = OcAgentNodeUtils.getNodeInfo(serviceName);
+
+    // First message must have Node set.
+    ExportTraceServiceRequest firstExportReq =
+        ExportTraceServiceRequest.newBuilder().setNode(node).build();
+
+    // Export will be handled in this thread.
+    @Nullable
+    StreamObserver<ExportTraceServiceRequest> exportRequestObserver =
+        initiateConnectionWithRetries(
+            stub, firstExportReq, exportResponseObserver, ServiceName.EXPORT);
+
+    // In a daemon thread, handle trace configurations that are beamed down by the agent, also
+    // reply to it with the applied configuration.
+    factory.newThread(new ConfigServiceClientWorker(stub, node)).start();
+
+    if (exportRequestObserver == null || currentConfigObserverRef.get() == null) {
+      return null;
+    }
+    return new ExportServiceClient(exportRequestObserver);
+  }
+
+  // Tries to initiate connection with Agent server with exponential backoff.
+  @Nullable
+  @VisibleForTesting
+  static <ReqT, RespT> StreamObserver<ReqT> initiateConnectionWithRetries(
+      TraceServiceStub stub,
+      ReqT request,
+      StreamObserver<RespT> responseObserver,
+      ServiceName serviceName) {
+    for (int i = 0; i < MAX_RETRY_TIMES; i++) {
+      try {
+        return initiateConnection(stub, request, responseObserver, serviceName);
+      } catch (StatusRuntimeException e) { // Only retry on gRPC StatusRuntimeException.
+        logger.warning(
+            String.format(
+                "Failed to connect to OC-Agent because of %s. Retried %d time(s).", e, i));
+        // Exponential backoff.
+        try {
+          Thread.sleep(0, getBackoffTimeInNanos(i));
```

Comment:
I think we should backoff with more than 1ms. Based on the definition of the Sleep the number of nanos has to be <1M.

Reply:
I removed exponential backoff since we'll retry every 5 minutes (or a custom time interval).